library/lib/view.cpp

```diff
@@ -627,4 +627,13 @@ void View::invalidate(bool immediate)
         this->dirty = true;
 }
 
+View* View::getChildViewAtTouch(float xpos, float ypos){
```

Comment:
This method could be made generic if we had a generic parent -> child relationship next to the child -> parent one. In that case we can just iterate over all children of the view tree and find the right one directly. We can then get rid of each reimplementation of getChidViewAtTouch.

Would that be inefficient?

Reply:
Sorry if I am misunderstanding you, but If I understood correctly you want there to be a function call that returns an iterable list of children that could be used to find the correct child?
In other words replace **getChildViewAtTouch** with **getChildren?**

Reply:
Remove getChildViewAtTouch in every View implementation and replace it with a loop that traverses the view tree, starting with the root flowing down to the children.

But to do that you need a generic parent -> child relation between views. Right now we only have parent -> child.

Reply:
commit fccfeda is my take on this.

Reply:
There is an Issue with TabFrame though, but I'm going to push another commit to fix that.

Reply:
I like the automated children detection in `setParent`. However the same view can be added twice - setParent should be idempotent. Could you use a data structure that doesn't allow duplicates ?

Also, the vector is never cleaned when the children are deleted so it looks like a hazard to me

Reply:
When you say "cleaned" what do you mean?

Reply:
As in "deleted children are not removed from the vector"

Reply:
Hmm, I see. I guess the problem is that there really isn't a system put in place that manages the relationship between child and parent views in a generalized way, and because of that it's really difficult to incorporate some sort of touch support without having to insert some specialization into many view classes so that it plays nicely with touches. What do you think?